app/src/common/shared/com/igalia/wolvic/browser/engine/SessionUtils.java

```diff
@@ -18,24 +19,30 @@ class SessionUtils {
 
     private static final String PREFERENCES_FILENAME = "fxr_config.yaml";
 
+    private static final String LAUNCH_IMMERSIVE_CONFIG = "  dom.vr.require-gesture: false\n  media.autoplay.default: 0";
+
     public static boolean isLocalizedContent(@Nullable String url) {
         return url != null && (url.startsWith("about:") || url.startsWith("data:"));
     }
 
     public static String prepareConfigurationPath(Context aContext) {
         File path = new File(aContext.getFilesDir(), PREFERENCES_FILENAME);
 
-        InputStream yaml = aContext.getResources().openRawResource(R.raw.fxr_config);
+        try (InputStream yaml = aContext.getResources().openRawResource(R.raw.fxr_config);
+             FileOutputStream outputStream = new FileOutputStream(path, false)) {
 
-        try (FileOutputStream outputStream = new FileOutputStream(path, false)) {
             int read;
             int bufferSize = 8192;
             byte[] bytes = new byte[bufferSize];
             while ((read = yaml.read(bytes)) != -1) {
                 outputStream.write(bytes, 0, read);
             }
+
+            if (aContext instanceof WidgetManagerDelegate && ((WidgetManagerDelegate) aContext).isLaunchImmersive()) {
```

Comment:
Can aContext by anything different from WidgetManagerDelegate? If not I prefer we use an assert here and then the type casting directly.

Reply:
Yes, that was actually the case before this PR because `LoginStorage` was the first one to call `EngineProvider.getOrCreateRuntime()` with the application's `Context`.

The reason for not having an assert is that `getOrCreateRuntime()` is called from different places and in principle we don't know which one will trigger this codepath.